src/domain/datasets/services/src/dataset_entry_service.rs

```diff
@@ -0,0 +1,285 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::sync::Arc;
+
+use dill::{component, interface, meta, Catalog};
+use init_on_startup::{InitOnStartup, InitOnStartupMeta};
+use internal_error::{InternalError, ResultIntoInternal};
+use kamu_accounts::{
+    AccountRepository,
+    DEFAULT_ACCOUNT_ID,
+    JOB_KAMU_ACCOUNTS_PREDEFINED_ACCOUNTS_REGISTRATOR,
+};
+use kamu_core::{
+    DatasetLifecycleMessage,
+    DatasetLifecycleMessageCreated,
+    DatasetLifecycleMessageDeleted,
+    DatasetLifecycleMessageRenamed,
+    DatasetRepository,
+    MESSAGE_PRODUCER_KAMU_CORE_DATASET_SERVICE,
+};
+use kamu_datasets::{DatasetEntry, DatasetEntryRepository};
+use messaging_outbox::{
+    MessageConsumer,
+    MessageConsumerMeta,
+    MessageConsumerT,
+    MessageConsumptionDurability,
+};
+use opendatafabric as odf;
+use opendatafabric::DatasetHandle;
+use time_source::SystemTimeSource;
+
+use crate::{JOB_KAMU_DATASETS_DATASET_ENTRY_SERVICE, MESSAGE_CONSUMER_KAMU_DATASET_ENTRY_SERVICE};
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[component(pub)]
+pub struct DatasetEntryServiceInitializationSkipper {}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[component(pub)]
+#[interface(dyn MessageConsumer)]
+#[interface(dyn MessageConsumerT<DatasetLifecycleMessage>)]
+#[meta(MessageConsumerMeta {
+    consumer_name: MESSAGE_CONSUMER_KAMU_DATASET_ENTRY_SERVICE,
+    feeding_producers: &[
+        MESSAGE_PRODUCER_KAMU_CORE_DATASET_SERVICE,
+    ],
+    durability: MessageConsumptionDurability::Durable,
+})]
+#[interface(dyn InitOnStartup)]
+#[meta(InitOnStartupMeta {
+    job_name: JOB_KAMU_DATASETS_DATASET_ENTRY_SERVICE,
+    depends_on: &[JOB_KAMU_ACCOUNTS_PREDEFINED_ACCOUNTS_REGISTRATOR],
+    requires_transaction: true,
+})]
```

Comment:
Consider implementing the indexing job as a separate component.
1) You would be able to avoid this skipper completely, just don't register the job in the DI, if conditoin is not met. Besides, you are not using any private implementation logic in the index, just public trait methods.
2) Only the indexing part needs a `dataset_repo` and `account_repository`. DI would instantiate the indexing job just once, unlike dataset entry service, which will be instantiated very often when handling dataset events in the outbox. At least the `dataset_repo` is quite heavy to instantiate, so if we can avoid it, we should.

Reply:
That's a cool suggestion, yes, that's what we should do

Done